designer/server/src/test/scala/pl/touk/nussknacker/ui/api/ManagementResourcesSpec.scala

```diff
@@ -387,28 +387,28 @@ class ManagementResourcesSpec extends AnyFunSuite with ScalatestRouteTest with F
   test("execute not implemented custom action") {
     createEmptyProcess(SampleProcess.processName)
     customAction(SampleProcess.processName, CustomActionRequest("not-implemented")) ~> check {
-      status shouldBe StatusCodes.NotImplemented
+      status shouldBe StatusCodes.BadRequest
     }
   }
 
   test("execute custom action with not allowed process status") {
     createEmptyProcess(SampleProcess.processName)
     customAction(SampleProcess.processName, CustomActionRequest("invalid-status")) ~> check {
-      status shouldBe StatusCodes.Forbidden
+      status shouldBe StatusCodes.BadRequest
     }
   }
 
-  test("should return 403 when execute custom action on archived process") {
+  test("should return 409 when execute custom action on archived process") {
     createArchivedProcess(SampleProcess.processName)
     customAction(SampleProcess.processName, CustomActionRequest("hello")) ~> check {
-      status shouldBe StatusCodes.Forbidden
+      status shouldBe StatusCodes.Conflict
```

Comment:
IMO forbidden was more descriptive than conflict. Maybe we should introduce another type of `EspError`. WDYT?

Reply:
In that case we should also change `case class ProcessIllegalAction ... with IllegalOperationError` which is translated by `EspErrorToHttp.espErrorToHttp` to `StatusCodes.Conflict`. Here I wanted custom action errors to be coherrent with `DeploymentServiceImpl.checkIfCanPerformActionOnProcess`. See also https://github.com/TouK/nussknacker/pull/4720/files#diff-a43c6577fba5055f9bbe2b11397d162fdbd97e7bbfd6fed52c04940276d48d6aR407

Reply:
Yep IMO, conflict wasn't good for the IllegalOperationError as well. Let's return forbidden as well.

Reply:
Done. Changed `ProcessIllegalAction` and removed redundant exceptions.